packages/main/tests/robot/test_excel.robot

```diff
@@ -23,14 +21,64 @@ Append Content To Sheet
     Should Be Equal    ${data}    ${content}
     Close Workbook
 
+Read Rows From Worksheet
+    [Arguments]    ${excel_file}
+    ${src} =    Set Variable    ${EXCELS}${/}${excel_file}
+    ${dest} =    Set Variable    ${OUTPUT_DIR}${/}${excel_file}
+    Copy File    ${src}    ${dest}
+    Open Workbook    ${dest}
+    ${rows}=    Read Worksheet    header=True
+    Close Workbook
+    [Return]    ${rows}
+
+Append Rows to Target
+    [Arguments]    ${rows}    ${target_filepath}    ${expected_empty_row}    ${formatting}=${False}
+    Open Workbook    ${target_filepath}
+    Append Rows to Worksheet    ${rows}    formatting_as_empty=${formatting}
+    Save Workbook
+    ${empty_row_number}=    Find Empty Row
+    Should Be Equal As Integers    ${expected_empty_row}    ${empty_row_number}
+    Close Workbook
 
 *** Tasks ***
 Test single row sheet
     # "Single" in this case acts like header for a 1x1 table.
     &{row} =    Create Dictionary    Single    Test
     @{content} =    Create List    ${row}
-
     Append Content To Sheet    one-row.xlsx    ${content}
     Append Content To Sheet    one-row.xls    ${content}
     Append Content To Sheet    empty.xlsx    ${content}
     Append Content To Sheet    empty.xls    ${content}
+
+Test appending content when formatted cell is considered as empty cell
+    ${target_file}=    Set Variable    ${OUTPUT_DIR}${/}target1.xlsx
+    Copy File    ${EXCELS}${/}data_template.xlsx    ${target_file}
+    ${rows1}=    Read Rows From Worksheet    data1.xlsx
+    Append Rows To Target    ${rows1}    ${target_file}    42    ${True}
+    ${rows2}=    Read Rows From Worksheet    data2.xlsx
+    Append Rows To Target    ${rows2}    ${target_file}    42    ${True}
+
+Test default appending content
+    ${target_file}=    Set Variable    ${OUTPUT_DIR}${/}target2.xlsx
+    Copy File    ${EXCELS}${/}data_template.xlsx    ${target_file}
+    ${rows1}=    Read Rows From Worksheet    data1.xlsx
+    Append Rows To Target    ${rows1}    ${target_file}    52
+    ${rows2}=    Read Rows From Worksheet    data2.xlsx
+    Append Rows To Target    ${rows2}    ${target_file}    62
+
+Test append rows to a target file
+    @{rows} =    Create List
+    FOR    ${counter}    IN RANGE    1    51
+        &{row} =    Create Dictionary
+        ...    Name    Cosmin
+        ...    Age    29
+        ...    E-mail    cosmin@robocorp.com
+        Append To List    ${rows}    ${row}
+    END
+    ${workbook} =    Set Variable    ${OUTPUT_DIR}${/}emails.xlsx
```

Comment:
this is what the end result is going to be named ? why it would need to be data_template instead of emails ?

Reply:
Not a big deal. Just that I've seen the header `name, type, price` in the Excel template and I though _emails.xlsx_ would be misleading here.
It can be _data.xlsx_ or _prices.xlsx_ as well, doesn't matter. (we can even keep _emails.xlsx_ just that the header in the template is not matching the values we insert)

I improved my testing example bot [here](https://github.com/cmin764/robots/blob/main/excel-files-app/tasks.robot#L188-L189):
```robotframework
    Append Rows to Worksheet    ${rows}    formatting_as_empty=${True}
    ...    header=${True}    start=${2}
```
to make sure I'm also matching the headers.